blaze-core/src/main/scala/org/http4s/blazecore/util/Http1Writer.scala

```diff
@@ -0,0 +1,28 @@
+package org.http4s
+package blazecore
+package util
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+import scala.concurrent._
+
+import fs2._
+import org.http4s.blaze.http.Headers
+import org.http4s.blaze.pipeline.TailStage
+import org.http4s.blaze.http.http20.NodeMsg._
+import org.http4s.util.StringWriter
+import org.http4s.util.chunk._
+
+trait Http1Writer extends EntityBodyWriter {
+  final def write(headerWriter: StringWriter, body: EntityBody): Task[Boolean] =
+    Task.fromFuture(writeHeader(headerWriter)).flatMap(_ => writeEntityBody(body))
```

Comment:
If the write of the headers fails nobody is going to execute the body, but maybe that's okay? It's not crazy to envision an `EntityBody` that is already holding onto resources like a socket connection.

Reply:
That's a good point.  I guess we need to drain the body on failure.